Library/Homebrew/utils/shared_audits.rb

```diff
@@ -12,6 +12,16 @@ module SharedAudits
 
   module_function
 
+  def eol_data(product, cycle)
+    @eol_data ||= {}
+    @eol_data["#{product}/#{cycle}"] ||= begin
+      out, _, status = Utils::Curl.curl_output("--location", "https://endoflife.date/api/#{product}/#{cycle}.json")
+      json = JSON.parse(out) if status.success?
+      json = nil if json&.dig("message")&.include?("Product not found")
```

Comment:
Wondering if we want some `opoo` or `odebug` here so e.g. a change to the API doesn't result in us silently not noticing this doesn't work any more?

Reply:
I can do that, what do you want to error on?

Reply:
@SMillerDev Feels like should definitely error on JSON failing to parse (which I think it maybe already does?). Not sure if the expected failure code is an unsuccessful status code or the product not found. Would be good to fail on whatever is not expected.

Reply:
Yeah, JSON parsing will fail. I believe the "Not Found" failure is both a 404 and a message actually.

Reply:
Yeh, can probably skip the message check then I think and fail on anything except a `404` status code?

Reply:
Do we have any reference for checking status codes? I can't find any.

Reply:
`curl_check_http_content` and `curl_http_content_headers_and_checksum` and `parse_curl_output` in `utils/curl` should provide some examples here.

Feel free to say "this is too involved, I can't be bothered" and that's fine 😁

Reply:
Yeah, that looks a lot more complicated than parsing JSON. Gonna keep it as-is